src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+        );
+
+        // Assert operator's signature is valid.
+        EIP1271SignatureUtils.checkSignature_EIP1271(operator, digestHash, operatorSignature.signature);
+        // Spend salt.
+        operatorSaltIsSpent[operator][operatorSignature.salt] = true;
+    }
+
+    /**
+     * @notice Calculates the digest hash to be signed by an operator to modify magnitude allocations
+     * @param operator The operator to allocate or deallocate magnitude for.
+     * @param allocations The magnitude allocations/deallocations to be made.
+     * @param salt A unique and single use value associated with the approver signature.
+     * @param expiry Time after which the approver's signature becomes invalid.
+     */
+    function calculateMagnitudeAllocationDigestHash(
+        address operator,
+        MagnitudeAllocation[] calldata allocations,
+        bytes32 salt,
+        uint256 expiry
+    ) public view returns (bytes32) {
+        return _calculateDigestHash(
+            keccak256(abi.encode(MAGNITUDE_ADJUSTMENT_TYPEHASH, operator, allocations, salt, expiry))
+        );
+    }
+
+    /// @notice Getter function for the current EIP-712 domain separator for this contract.
+    /// @dev The domain separator will change in the event of a fork that changes the ChainID.
+    function domainSeparator() public view override returns (bytes32) {
+        return _calculateDomainSeparator();
+    }
+
+    /// @notice Internal function for calculating the current domain separator of this contract
+    function _calculateDomainSeparator() internal view returns (bytes32) {
+        if (block.chainid == ORIGINAL_CHAIN_ID) {
+            return _DOMAIN_SEPARATOR;
+        } else {
+            return keccak256(abi.encode(DOMAIN_TYPEHASH, keccak256(bytes("EigenLayer")), block.chainid, address(this)));
+        }
+    }
+
+    /// @notice Returns an EIP-712 encoded hash struct.
+    function _calculateDigestHash(
+        bytes32 structHash
+    ) internal view returns (bytes32) {
+        return keccak256(abi.encodePacked("\x19\x01", _calculateDomainSeparator(), structHash));
+    }
+
+    /// @dev Returns an `OperatorSet` encoded into a 32-byte value.
+    /// @param operatorSet The `OperatorSet` to encode.
+    function _encodeOperatorSet(
+        OperatorSet memory operatorSet
+    ) internal pure returns (bytes32) {
+        return bytes32(abi.encodePacked(operatorSet.avs, uint96(operatorSet.operatorSetId)));
+    }
+
+    /// @dev Returns an `OperatorSet` decoded from an encoded 32-byte value.
+    /// @param encoded The encoded `OperatorSet` to decode.
+    /// @dev Assumes `encoded` is encoded via `_encodeOperatorSet(operatorSet)`.
+    function _decodeOperatorSet(
+        bytes32 encoded
+    ) internal pure returns (OperatorSet memory) {
+        return OperatorSet({
+            avs: address(uint160(uint256(encoded) >> 96)),
+            operatorSetId: uint32(uint256(encoded) & type(uint96).max)
```

Comment:
what's the `& type(uint96).max` part of this operation doing here? wouldn't just straight casting this to a uint32 do the same thing? is this just to make extra sure the avs address is masked out?

Reply:
In _encodeOperatorSet, the uint32 is casted to uint96 so that the encodePacked result yields a bytes32 output. I believe yes, the mask is just to properly ensure we are taking all the non-address bits.